api.go

```diff
@@ -200,7 +202,217 @@ func (r *Router) RouterCallImpl(ctx context.Context,
 	}
 }
 
-// todo: router_map_callrw
+type MapCallMetaData struct {
+	Resp  []interface{}
+	Error error
+
+	replicaset *Replicaset
+	future     *tarantool.Future
+}
+
+func (r *Router) getMapCallSelectCases(req tarantool.Request) []reflect.SelectCase {
+	metaDatas := make([]<-chan *MapCallMetaData, 0)
+	for _, replicaset := range r.idToReplicaset {
+		ch := make(chan *MapCallMetaData)
+		go func(replicaset *Replicaset, ch chan *MapCallMetaData) {
+			conn := replicaset.conn
+
+			future := conn.Do(req, pool.RW)
+
+			resp, err := future.Get()
+
+			metaData := &MapCallMetaData{
+				Resp:       resp,
+				Error:      err,
+				future:     future,
+				replicaset: replicaset,
+			}
+
+			ch <- metaData
+		}(replicaset, ch)
+
+		metaDatas = append(metaDatas, ch)
+	}
+
+	cases := make([]reflect.SelectCase, len(metaDatas))
+	for i, metaData := range metaDatas {
+		cases[i] = reflect.SelectCase{
+			Dir:  reflect.SelectRecv,
+			Chan: reflect.ValueOf(metaData),
+		}
+	}
+
+	return cases
+}
+
+func (r *Router) callStorageUnref(refID int64) {
```

Comment:
please write a small comment for all functions

Reply:
u can use just a link to tarantool doc  with ref https://www.tarantool.io/ru/doc/latest/book/admin/vshard_admin/#bucket-ref

Reply:
put comment